core/src/cluster_slots_service.rs

```diff
@@ -100,12 +98,25 @@ impl ClusterSlotsService {
             lowest_slot_elapsed.stop();
             let mut process_cluster_slots_updates_elapsed =
                 Measure::start("process_cluster_slots_updates_elapsed");
+
             if let Some(slots) = slots {
-                Self::process_cluster_slots_updates(
-                    slots,
-                    &cluster_slots_update_receiver,
-                    &cluster_info,
-                );
+                let node_id = cluster_info.id();
+                let my_stake = epoch_specs
+                    .current_epoch_staked_nodes()
+                    .get(&node_id)
+                    .cloned()
+                    .unwrap_or_default();
```

Comment:
avoid smearing over potential unexpected error scenarios with dubious generics (`Default` is a bug). this would have been more obvious as
```rust
let maybe_my_stake = epoch_specs
    .current_epoch_staked_nodes()
    .get(&node_id)
    .cloned();
match maybe_my_stake {
    Some(my_stake) if my_stake > 0 => {
        ...process
    }
    () => {
        ...drain
    }
}
```

Reply:
Arguably, `unwrap_or(0)` would work even better here. Agree that default can do surprising things. Should we add recommendations to avoid using default() where reasonable?